packages/firestore/src/model/path.ts

```diff
@@ -215,6 +215,15 @@ export class ResourcePath extends BasePath<ResourcePath> {
     return this.canonicalString();
   }
 
+  /**
+   * Returns a string representation of this path
+   * where each path segment has been encoded with
+   * `encodeURIComponent`.
+   */
+  toUriEncodedString(): string {
```

Comment:
IMO, `toUriEncodedString()` shouldn't be on ResourcePath since it's an implementation detail specific to RestConnection, especially since it's only called once in the entire SDK. Consider moving this logic into RestConnection.

Reply:
I considered that, but I felt more strongly that the structure of the path and how to URI encode the path should be the responsibility of the Path class. IMO, the connection shouldn't be responsible for knowing how to break up a path and encode its parts.